tests/integrations/celery/test_celery.py

```diff
@@ -616,3 +616,17 @@ def example_task():
         pytest.fail("Calling `apply_async` without arguments raised a TypeError")
 
     assert result.get() == "success"
+
+
+def test_retry_count(init_celery, capture_events):
+    celery = init_celery(enable_tracing=True)
+    events = capture_events()
+
+    @celery.task()
+    def task(): ...
+
+    task.apply_async()
+
+    (event,) = events
+    (span,) = event["spans"]
+    assert span["data"]["messaging.message.retry.count"] == 0
```

Comment:
I am unsure how we can test a retry count >0, or whether it is even possible with our current test setup

Reply:
Apparently we can, added a nonzero test in [edf4c0a](https://github.com/getsentry/sentry-python/pull/2971/commits/edf4c0a9ae2a396957697c0075ee047a40d38706)